src/libs/io/adapter/dbus/dbus.c

```diff
@@ -0,0 +1,274 @@
+/**
+ * @file
+ *
+ * @brief I/O Adapter for D-Bus.
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+#include <kdbhelper.h>
+#include <kdbio_adapter_dbus.h>
+#include <kdblogger.h>
+
+#include <stdlib.h>
+#include <string.h>
+
+
+typedef struct _ElektraIoAdapterDbusHandle
+{
+	DBusConnection * connection;
+	ElektraIoInterface * ioBinding;
+	ElektraIoIdleOperation * dispatchIdle;
+} _ElektraIoAdapterDbusHandle;
+
+typedef struct DbusAdapterWatchInfo
+{
+	_ElektraIoAdapterDbusHandle * private;
+	DBusWatch * watch;
+} DbusAdapterWatchInfo;
+
+static int dbusToFlags (int dbus)
+{
+	// fprintf(stderr, "dbusToFlags\n");
+	int flags = 0;
+	if (dbus & DBUS_WATCH_READABLE)
+	{
+		flags |= ELEKTRA_IO_READABLE;
+	}
+	if (dbus & DBUS_WATCH_WRITABLE)
+	{
+		flags |= ELEKTRA_IO_WRITABLE;
+	}
+	// from docs: "all watches implicitly include a watch for hangups, errors, and other exceptional conditions."
+	// TODO add DBUS_WATCH_ERROR and DBUS_WATCH_ERROR to flags
+	return flags;
+}
+
+static void dbusWrapperFree (void * memory)
+{
+	// fprintf(stderr, "dbusWrapperFree\n");
```

Comment:
Please use logger here.

Reply:
These were some leftovers from testing - I removed the fprintf statements.